engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorContextProperties.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine;
+
+import java.time.Duration;
+
+import javax.validation.ClockProvider;
+import javax.validation.MessageInterpolator;
+import javax.validation.TraversableResolver;
+import javax.validation.Validator;
+
+import org.hibernate.validator.internal.engine.resolver.TraversableResolvers;
+import org.hibernate.validator.internal.util.ExecutableParameterNameProvider;
+import org.hibernate.validator.spi.scripting.ScriptEvaluatorFactory;
+
+/**
+ * Context object keeping storing all possible validator configurable properties.
+ * There should be one per {@link Validator} instance.
+ *
+ * @author Marko Bekhta
+ */
+public class ValidatorContextProperties {
+
+	/**
+	 * The default message interpolator for this factory.
+	 */
+	private final MessageInterpolator messageInterpolator;
+
+	/**
+	 * The default traversable resolver for this factory.
+	 */
+	private final TraversableResolver traversableResolver;
```

Comment:
As mentioned earlier, I think it would be a mistake to include this one. Let's keep it out of it to make it clear it can be validation scoped.